File: CS/SecuritySetState.Module/BusinessObjects/MyTask.cs
using System;
using System.Linq;
using System.Text;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using System.ComponentModel;
using DevExpress.ExpressApp.DC;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Editors;
using DevExpress.Persistent.Base.General;
using DevExpress.ExpressApp.SystemModule;

namespace SecuritySetState.Module.BusinessObjects {
    [DefaultClassOptions]
    public class MyTask : BaseObject {
        public MyTask(Session session)
            : base(session) {
        }
        public override void AfterConstruction() {
            base.AfterConstruction();
        }
        string subject;
        public string Subject {
            get {
                return subject;
            }
            set {
                SetPropertyValue(nameof(Subject), ref subject, value);
            }
        }
        Contact assignedTo;
        [Association("Contact-Tasks")]
        public Contact AssignedTo {
            get {
                return assignedTo;
            }
            set {
                SetPropertyValue(nameof(AssignedTo), ref assignedTo, value);
            }
        }
        bool isActive;
        public bool IsActive {
            get {
                return isActive;
            }
            set {
                SetPropertyValue(nameof(IsActive), ref isActive, value);
            }
        }
        Priority priority;
        public Priority Priority {
            get {
                return priority;
            }
            set {
                SetPropertyValue(nameof(Priority), ref priority, value);
            }
        }
    }
    public enum Priority {
        [ImageName("State_Priority_Low")]
        Low = 0,
        [ImageName("State_Priority_Normal")]
        Normal = 1,
        [ImageName("State_Priority_High")]
        High = 2
    }
}

File: VB/SecuritySetState.Module/BusinessObjects/MyTask.vb
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace SecuritySetState.Module.BusinessObjects

    <DefaultClassOptions>
    Public Class MyTask
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub

        Private subjectField As String

        Public Property Subject As String
            Get
                Return subjectField
            End Get

            Set(ByVal value As String)
                SetPropertyValue(NameOf(MyTask.Subject), subjectField, value)
            End Set
        End Property

        Private assignedToField As Contact

        <Association("Contact-Tasks")>
        Public Property AssignedTo As Contact
            Get
                Return assignedToField
            End Get

            Set(ByVal value As Contact)
                SetPropertyValue(NameOf(MyTask.AssignedTo), assignedToField, value)
            End Set
        End Property

        Private isActiveField As Boolean

        Public Property IsActive As Boolean
            Get
                Return isActiveField
            End Get

            Set(ByVal value As Boolean)
                SetPropertyValue(NameOf(MyTask.IsActive), isActiveField, value)
            End Set
        End Property

        Private priorityField As Priority

        Public Property Priority As Priority
            Get
                Return priorityField
            End Get

            Set(ByVal value As Priority)
                SetPropertyValue(NameOf(MyTask.Priority), priorityField, value)
            End Set
        End Property
    End Class

    Public Enum Priority
        <ImageName("State_Priority_Low")>
        Low = 0
        <ImageName("State_Priority_Normal")>
        Normal = 1
        <ImageName("State_Priority_High")>
        High = 2
    End Enum
End Namespace
